### PR TITLE
vgg-push

### PR DESCRIPTION
**Pull request by zjuter0126**

### CODE REVIEW

File: tinyms/model/vgg.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+cfgs = {
+    'A': [64, 'M', 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M'],
+    'B': [64, 64, 'M', 128, 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M'],
+    'D': [64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'M', 512, 512, 512, 'M', 512, 512, 512, 'M'],
+    'E': [64, 64, 'M', 128, 128, 'M', 256, 256, 256, 256, 'M', 512, 512, 512, 512, 'M', 512, 512, 512, 512, 'M'],
+}
+
+
+def vgg11(class_num=10):
```
Comment (leonwanghui) [inline tinyms/model/vgg.py:81]: Please use `*kwargs` to extend the arguments

File: tinyms/model/vgg.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+
+import numpy as np
+
+from tinyms import layers, Tensor
+from tinyms.layers import BatchNorm2d, ReLU, Conv2d, MaxPool2d, \
```
Comment (leonwanghui) [inline tinyms/model/vgg.py:19]: If u have already imported `layers` module, it should not be necessary to import all these concrete operators.

File: tinyms/model/vgg.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    return Tensor(init_value)
+
+
+def _conv3x3(in_channel, out_channel, stride=1):
+    weight_shape = (out_channel, in_channel, 3, 3)
+    weight = _weight_variable(weight_shape)
+    return layers.Conv2d(in_channel, out_channel,
+                         kernel_size=3, stride=stride, padding=0, pad_mode='same', weight_init=weight)
+
+class VGG(layers.Layer):
```
Comment (leonwanghui) [inline tinyms/model/vgg.py:32]: Please add the annotation of `VGG` class, so that others could know how to use it.

File: tests/st/vgg16.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+if __name__ == '__main__':
+    args_opt = parse_args()
+    context.set_context(mode=context.GRAPH_MODE, device_target=args_opt.device_target)
+
+    # download cifar10 dataset
+    if not args_opt.dataset_path:
+        args_opt.dataset_path = download_dataset('cifar10')
+    # build the network
+    net = vgg16(args_opt.num_classes)
```
Comment (leonwanghui) [inline tests/st/vgg16.py:79]: This should be `class_num=args_opt.num_classes`, or else it would fail to run.

**Review by leonwanghui — CHANGES_REQUESTED** [review]

File: tests/st/vgg16.py (hunk 4 above)
Comment (zjuter0126) [reply tests/st/vgg16.py]: I have correct this error and add batch_norm to choice whether to use bn

**Review by leonwanghui — APPROVED** [review]
LGTM

**Pull request merged**